trio/_socket.py

```diff
@@ -1,39 +1,37 @@
 import os as _os
-import socket as _stdlib_socket
 import sys as _sys
+import importlib as _importlib
 from functools import wraps as _wraps
 
+_stdlib_socket = _importlib.import_module('socket')
+
 import idna as _idna
 
 from . import _core
 from ._threads import run_sync_in_worker_thread
 from ._util import fspath
 
-__all__ = []
-
-################################################################
-# misc utilities
-################################################################
-
-
-def _reexport(name):
-    globals()[name] = getattr(_stdlib_socket, name)
-    __all__.append(name)
-
+from socket import (
+    gaierror,
+    herror,
+    gethostname,
+    ntohs,
+    htonl,
+    htons,
+    inet_aton,
+    inet_ntoa,
+    inet_pton,
+    inet_ntop,
+)
 
-def _add_to_all(obj):
-    __all__.append(obj.__name__)
-    return obj
+try:
+    from socket import (
+        if_nameindexif_nametoindex, if_indextoname, sethostname
+    )
```

Comment:
I thought I had fixed this already but I'll take a look again. 

But this typo shows the weakness of this approach. I had to put these four into a try catch block because on py3.5 they are not there. Actually I should put each of the names in a try except block of its own but that would make the code even uglier.

So we do not only have a constant version/os dependency but also for functions.

I can see you are somehow reluctant towards a star import with overwriting which I now would prefer but I still do not know why exactly. Now you have about 25 years more experience than me so this is very likely well founded, I'd just like to understand why.